server/src/main/java/com/saralein/server/parameters/ParameterParser.java

```diff
@@ -0,0 +1,59 @@
+package com.saralein.server.parameters;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ParameterParser {
+    public Map<String, String> parse(String uri) throws Exception {
+        Map<String, String> params = new HashMap<>();
+        String[] splitUriAndParams = splitUriAndParams(uri);
+
+        if (splitUriAndParams.length > 1) {
+            String rawParams = splitUriAndParams[1];
+            params = formatParamsIfValid(rawParams);
+        }
+
+        return params;
+    }
```

Comment:
🤔 Do these exceptions make the server respond with an error code?

Reply:
Yeah.  Perhaps my PR is too small now and I should included the decoder and controller.

Reply:
Yeah, it might help to see it all working together.

Reply:
OK, all the pieces should be there now.